packages/lit-html/src/directives/ref.ts

```diff
@@ -30,16 +30,20 @@ interface RefInternal {
 
 // When callbacks are used for refs, this map tracks the last value the callback
 // was called with, for ensuring a directive doesn't clear the ref if the ref
-// has already been rendered to a new spot
-const lastElementForCallback: WeakMap<Function, Element | undefined> =
-  new WeakMap();
+// has already been rendered to a new spot. It is double-keyed on both the
+// callback and the context (options.host), since we auto-bind class methods to
+// options.host.
+const lastElementForCallback: WeakMap<
```

Comment:
It seems like the WeakMap setup would make more sense if it was keyed off the host first.

Reply:
Done